ert_shared/client/session.py

```diff
@@ -0,0 +1,58 @@
+import os
+import requests
+import json
+
+from urllib.parse import urljoin
+from typing import Any, Tuple, Optional
+
+
+class Session(requests.Session):
+    def __init__(self):
+        super().__init__()
+        self._base_url: Optional[str] = None
+        self._headers: Optional[Tuple] = ()
+
+        # Should the storage_server.json also be used?
+        # If so is there order of prefence
+        if "ERT_STORAGE_CONNECTION_STRING" in os.environ:
+            try:
+                self._connection_info = json.loads(
+                    os.environ["ERT_STORAGE_CONNECTION_STRING"]
+                )
+            except json.JSONDecodeError:
+                raise RuntimeError("Invalid Storage Connection configuration")
+
+        if {"urls", "authtoken"} <= self._connection_info.keys():
```

Comment:
Cool! I haven't seen this pattern before for checking whether a set of keys is contained in a different set. :)